ballerina/rag.bal

```diff
@@ -0,0 +1,167 @@
+    #
+    # + vectorStore - The vector store for embedding persistence
+    # + embeddingModel - The embedding provider for generating vector representations
+    public isolated function init(VectorStore vectorStore, EmbeddingProvider embeddingModel) {
+        self.vectorStore = vectorStore;
+        self.embeddingModel = embeddingModel;
+        self.retriever = new VectorRetriever(vectorStore, embeddingModel);
+    }
+
+    # Indexes a collection of chunks.
+    # Converts each chunk to an embedding and stores it in the vector store,
+    # making the chunk searchable through the retriever.
+    #
+    # + chunks - The array of chunk to index
+    # + return - An `ai:Error` if indexing fails; otherwise, `nil`
+    public isolated function ingest(Chunk[] chunks) returns Error? {
+        VectorEntry[] entries = [];
+        foreach Chunk chunk in chunks {
+            Embedding embedding = check self.embeddingModel->embed(chunk);
+            entries.push({embedding, chunk});
+        }
+        check self.vectorStore.add(entries);
+    }
+
+    # Retrieves relevant chunk for the given query.
+    #
+    # + query - The text query to search for
+    # + filters - Optional metadata filters to apply during retrieval
+    # + return - An array of matching chunks with similarity scores, or an `ai:Error` if retrieval fails
+    public isolated function retrieve(string query, MetadataFilters? filters = ()) returns QueryMatch[]|Error {
+        return self.retriever.retrieve(query, filters);
+    }
+}
+
+# Creates a default model provider based on the provided `wso2ProviderConfig`.
+# + return - A `Wso2ModelProvider` instance if the configuration is valid; otherwise, an `ai:Error`.
+public isolated function getDefaultModelProvider() returns Wso2ModelProvider|Error {
+    Wso2ProviderConfig? config = wso2ProviderConfig;
+    if config is () {
+        return error Error("The `wso2ProviderConfig` is not configured correctly."
+        + " Ensure that the WSO2 model provider configuration is defined in your TOML file.");
+    }
+    return new Wso2ModelProvider(config.serviceUrl, config.accessToken);
+}
+
+isolated function getDefaultKnowledgeBase() returns VectorKnowledgeBase|Error {
+    Wso2ProviderConfig? config = wso2ProviderConfig;
+    if config is () {
+        return error Error("The `wso2ProviderConfig` is not configured correctly."
+        + " Ensure that the WSO2 model provider configuration is defined in your TOML file.");
+    }
+    EmbeddingProvider|Error wso2EmbeddingProvider = new Wso2EmbeddingProvider(config.serviceUrl, config.accessToken);
+    if wso2EmbeddingProvider is Error {
+        return error Error("Error creating default vector knowledge base");
+    }
+    InMemoryVectorStore|Error inMemoryVectorStore = check new InMemoryVectorStore();
+    if inMemoryVectorStore is Error {
+        return error Error("Error creatign default inMemoryVector store", inMemoryVectorStore);
```

Comment:
Typo in creatign.

Also, in the error message, complete terms like in-memory vector store generally make more sense, unless it is a reference to something in the code.